frontends/election-manager/src/app.test.tsx

```diff
@@ -455,9 +456,10 @@ test('tabulating CVRs', async () => {
   );
 
   getByText('View Unofficial Full Election Tally Report');
-  fireEvent.click(getByText('Mark Tally Results as Official…'));
-  getByText('Mark Unofficial Tally Results as Official Tally Results?');
   fireEvent.click(getByText('Mark Tally Results as Official'));
+  getByText('Mark Unofficial Tally Results as Official Tally Results?');
+  const modal = await screen.findByRole('alertdialog');
```

Comment:
Ooh interesting, so as far as I know, `findByRole` doesn't actually return the found element. It just throws an error when it can't find something. To actually get and use the found element, you typically have to use one of the `getBy` queries: https://testing-library.com/docs/queries/about#types-of-queries

Would be curious to understand why the test is still passing!

Reply:
Actually, I take that back! Looks like it does return the element -- that's what the returned promise resolves to in the success case. (It wasn't completely clear in the docs what the returned promise resolved to)

Reply:
My reading of the documentation was that `findBy...` returns a promise of the element. I thought this was the case based on the chart in the docs:
![Screen Shot 2022-05-09 at 11 59 07 AM](https://user-images.githubusercontent.com/37960853/167479672-3f90be35-634c-40d1-9273-e20c435f983d.png)


Also, the Arlo example also uses `findBy...`: https://github.com/votingworks/arlo/blob/e2a58b5a557ef09c0fffd77385a9f3ed72e87f62/client/src/components/AuditBoard/AuditBoardView.test.tsx#L269-L281

Is there a difference between how it's being used in the Arlo case and this one?

Reply:
Ok, great! I was't sure about the comparative advantages of one or the other, so I emulated the way Arlo did it.